### PR TITLE
FPGA JTAG documentation updates

### PR DESCRIPTION
**Pull request by jlmahowa-amd**

### CODE REVIEW

File: hw-latest/fpga/FPGA_Instructions.md [hunk 1]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+    - `openocd.exe -f ~/openocd_caliptra.txt`
+1. Connect client(s) for debug
+    - GDB: Port 3333
+    - Telnet: Port 4444
+
+#### Caliptra SoC interface registers ####
+Over Telnet connection to OpenOCD: `riscv.cpu riscv dmi_read [addr]`
+
+#### JTAG Testing ####
+Both OpenOCD and GDB
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:143]: Recommend `Both OpenOCD and GDB require the following:`

File: hw-latest/fpga/FPGA_Instructions.md [hunk 2]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
 1. Execute test targeting fpga_realtime
     - `CPTRA_UIO_NUM=4 cargo test -p caliptra-test --features=fpga_realtime smoke_test`
 
-### JTAG debug [Doc WIP]
-JTAG support has been enabled using OpenOCD. The security state debug_locked parameter must be FALSE for JTAG to be accessible.
-- `openocd.exe -f ~/openocd_caliptra.txt`
+### JTAG debug
+Requirements
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:124]: add colon after `Requirements`

File: hw-latest/fpga/FPGA_Instructions.md [hunk 3]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+    - GDB: Port 3333
+    - Telnet: Port 4444
+
+#### Caliptra SoC interface registers ####
+Over Telnet connection to OpenOCD: `riscv.cpu riscv dmi_read [addr]`
+
+#### JTAG Testing ####
+Both OpenOCD and GDB
+- Ensure JTAG port is accessible when debug_locked == true or lifecycle == manuf, inaccessible otherwise.
+- Read access to ROM space using 8, 16, 32, and 64 bit reads
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:145]: add period at the end of all items in the lists in this section

File: hw-latest/fpga/FPGA_Instructions.md [hunk 4]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+- Read and write access to DCCM using 8, 16, 32, and 64 bit accesses
+- Access to ICCM using 32 and 64 bit reads, 32 bit writes.
+- Access to VEER core registers
+- HW and SW breakpoints
+- Watchpoints on DCCM and Caliptra registers
+ 
+GDB
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:152]: `GDB`-->`GDB requirements:`

File: hw-latest/fpga/FPGA_Instructions.md [hunk 5]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+- Basic commands all work (step, next, frame, info, bt, ni, si, etc)
+- Finish sometimes does nothing, but stepping usually resolves this behavior.
+ 
+OpenOCD
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:156]: `OpenOCD`-->`OpenOCD requirements:`

File: hw-latest/fpga/FPGA_Instructions.md [hunk 6]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+- Basic commands (reg, step, resume, etc)
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:157]: add period after `etc`

File: hw-latest/fpga/FPGA_Instructions.md [hunk 7]
```diff
@@ -122,16 +120,41 @@ Serial port settings for connection over USB.
...
+1. Invoke OpenOCD server
+    - `openocd.exe -f ~/openocd_caliptra.txt`
+1. Connect client(s) for debug
+    - GDB: Port 3333
+    - Telnet: Port 4444
+
+#### Caliptra SoC interface registers ####
+Over Telnet connection to OpenOCD: `riscv.cpu riscv dmi_read [addr]`
+
+#### JTAG Testing ####
```
Comment (steph-morton) [inline hw-latest/fpga/FPGA_Instructions.md:142]: lowercase `testing` in heading

Note: could do same throughout the doc (lowercase subsequent words in headings) but ok if out of scope for this PR.

**Review by steph-morton — COMMENTED** [review]
editorial feedback

**Review by FerralCoder — APPROVED** [review]

**Pull request merged**